contracts/PeeranhaToken.sol

```diff
@@ -112,19 +190,84 @@ contract PeeranhaToken is ERC20Upgradeable, ERC20PausableUpgradeable, ERC20Cappe
     return userReward;
   }
 
-  function reduceRewards(uint256 rewardWeek, uint16 period) private view returns(uint256) {
-    uint16 countReduce = period / 52;
+  function getPoolTokens(uint16 period) external view returns(uint256) {
+    return poolTokens[period];
+  }
 
-    for (uint16 i = 0; i < countReduce; i++) {
-      rewardWeek = (rewardWeek * 93) / 100;
+  function getUserBoostPeriods(address user) external view returns (uint16[] memory) {
+    TokenLib.UserBoost storage userBoost = weekUserBoost.weekUserBoost[user];
+    return userBoost.updateUserBoostInPeriod;
+  }
+
+  function getBoostPeriods() external view returns (uint16[] memory) {
+    return boostContainer.updateBoostInPeriod;
+  }
+
+  function getBoost(address user, uint16 period, int32 rating) public override view returns (uint256) {    // name
+    uint256 averageStake = getAverageStake(period);
+    uint256 userStake = getUserStake(user, period);
+    uint256 boost;
+
+    if (averageStake == 0 || userStake == 0) return uint256(rating) * TokenLib.getRewardCoefficient() * FRACTION;
```

Comment:
Where is this rule coming from? That returns a huge value for boost. 
Boost must be `1` no one set any boost.